concordium-sdk/src/main/java/com/concordium/sdk/transactions/UpdateCredentialKeys.java

```diff
@@ -0,0 +1,63 @@
+package com.concordium.sdk.transactions;
+
+import com.concordium.sdk.types.UInt64;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.val;
+
+import java.nio.ByteBuffer;
+
+
+/**
+ * Update signing keys of a specific credential.
+ */
+@Getter
+@EqualsAndHashCode(callSuper = true)
+@ToString
+public final class UpdateCredentialKeys extends Payload {
+    /**
+     * Id of the credential whose keys are to be updated.
+     */
+    private final CredentialRegistrationId credentialRegistrationID;
+
+    /**
+     *  The new public keys
+     */
+    private final CredentialPublicKeys keys;
+
+    private final UInt64 maxEnergyCost;
```

Comment:
This should not be necessary. It is possible to pre compute the cost of this type of transaction beforehand. 
See https://github.com/Concordium/concordium-base/blob/a4c5fa5349ee9c5b775dfea55d7e5f8e322be347/rust-src/concordium_base/src/transactions.rs#L1762

Reply:
Changed